backend/src/org/commcare/data/xml/DataModelPullParser.java

```
@@ -135,6 +136,9 @@ private void parseBlock(String root) throws InvalidStructureException, IOExcepti
                 if (!failfast) {
                     try {
                         transaction.parse();
+                    } catch(StorageFullException e){
```

By "Try forever" you just mean that it fails silently, right?